scripts/test/system.sh

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-set -em -o pipefail
```

Comment:
I don't think it's a good idea to drop these, if a command now fails it wont fail the test and we wont know that something broke.

Instead you could define a function that calls `_subctl` and expects it to fail.

For example, I broke the tests but they still passed:
```bash
::group::Running 'subctl uninstall -y --context cluster2 --fdsafs'
[06:32:07.131] [dir=subctl; cl=cluster2; fn=_subctl]$ /go/src/github.com/submariner-io/subctl/cmd/bin/subctl uninstall -y --context cluster2 --fdsafs
Error: unknown flag: --fdsafs
Usage:
  subctl uninstall [flags]

Flags:
      --as string                      Username to impersonate for the operation
...
  -y, --yes                            automatically answer yes to confirmation prompt

unknown flag: --fdsafs
[06:32:07.151] [dir=subctl; cl=cluster2; fn=_subctl]$ ret=1
::endgroup::
[06:32:07.152] [dir=subctl; cl=cluster2; fn=_subctl]$ return 1
...
Successfully remade target file 'system-test'.
[mkolesni@mkolaptop subctl ((69276fc...) *)]$ echo $?
0
```

Reply:
The problem is that `_subctl` ignored errors previously anyway. Fixing `_subctl` so that it no longer ignores errors, which I need so that the tests can actually verify that certain forms of `subctl` fail, causes the overall system test to fail earlier because `subctl` exits with an error in certain “normal” cases anyway.

Since `set -e -o pipefail` was effectively meaningless, I removed it; we can add it back later, once we have a consistent set of rules for `subctl`’s exit code.

Reply:
I don't understand what you mean by
> `_subctl` ignored errors previously anyway

and
> Since `set -e -o pipefail` was effectively meaningless

It didn't ignore errors, it would fail the tests due to any error, eg try to apply this and run the system tests:

```patch
diff --git a/scripts/test/system.sh b/scripts/test/system.sh
index 69a4e9c..62d470d 100755
--- a/scripts/test/system.sh
+++ b/scripts/test/system.sh
@@ -133,7 +133,7 @@ deploy_env_once
 
 # Test subctl show invocations
 
-_subctl show all | tee /dev/stderr | (sponge ||:) | grep -q 'Cluster "cluster2"'
+_subctl show all --sdfsdf | tee /dev/stderr | (sponge ||:) | grep -q 'Cluster "cluster2"'
 # Single-context variants don't say 'Cluster "foo"', check what cluster is considered local
 _subctl show all --context cluster1 | tee /dev/stderr | (sponge ||:) | grep -qv 'cluster2.*local'
 _subctl show all --context cluster2 | tee /dev/stderr | (sponge ||:) | grep -q 'cluster2.*local'
```

Reply:
Ah, I was barking up the wrong tree, it’s not that `_subctl` ignores errors, it’s the combination of functions and `&&` again :-/.

Reply:
Yes that one is indeed problematic but it doesn't have anything to do with the `set -e`..

You can make this change to ensure it (as we don't care about the specific exit code):
```suggestion
    "${DAPPER_SOURCE}"/cmd/bin/subctl "$@" || exit 1
```

GHA will be fine without a closing `::endgroup::` anyway, and the code will be tidier :)